core/src/Threads/Kokkos_ThreadsExec.cpp

```diff
@@ -794,6 +794,57 @@ int Threads::impl_thread_pool_rank() {
 const char *Threads::name() { return "Threads"; }
 } /* namespace Kokkos */
 
+namespace Impl {
+
+int g_threads_space_factory_initialized =
+    initialize_space_factory<ThreadsSpaceInitializer>("050_Threads");
+
+void ThreadsSpaceInitializer::initialize(const InitArguments &args) {
+  const int num_threads = args.num_threads;
+  const int use_numa    = args.num_numa;
+  if (std::is_same<Kokkos::Threads, Kokkos::DefaultExecutionSpace>::value ||
+      std::is_same<Kokkos::Threads,
+                   Kokkos::HostSpace::execution_space>::value) {
```

Comment:
please change this to a `static_assert` or just remove it unless we're going to do something with this `if`/`else`.

Reply:
See comment in #3443 we will do this separately after the fact so we can get this progressed.